polkit/authority.go

```diff
@@ -49,7 +49,7 @@ func checkAuthorization(subject authSubject, actionId string, details map[string
 	err = authority.Call(
 		"org.freedesktop.PolicyKit1.Authority.CheckAuthorization", 0,
 		subject, actionId, details, flags, "").Store(&result)
-	if err != nil && !result.IsAuthorized {
+	if err != nil || !result.IsAuthorized {
 		if result.IsChallenge {
 			err = ErrInteraction
 		} else if result.Details["polkit.dismissed"] != "" {
```

Comment:
I think we'll get a panic here if `err` is non-nil, also we override the actual error with our own. The whole check should be:
```
if err != nil {
    return false, err
}
if !result.IsAuthorized {
    ...
}
```
I'll push a patch on top of this commit if you don't mind.

Reply:
Sure, I don't mind!

Reply:
Aaand pushed.